### ISSUE TITLE
Current transaction builders make adding links tricky

### ISSUE THREAD

**Issue by librarianmage** [redstreet/beancount_reds_importers#102]
Each `TransactionBuilder` currently initializes its transactions with `data.EMPTY_SET`, which is a `frozenset`. As such, adding links can’t be done in place, but the current framework only provides in-place manipulation hooks.

**Comment by librarianmage** [comment]
perhaps the parameter could be `self.get_links() or data.EMPTY_SET` or `set(self.get_links()) or data.EMPTY_SET` or something

**Comment by redstreet** [comment]
Agree, and `self.get_links()` matches what we do with `self.get_tags()`. Feel free to submit a PR. Apologies for the very late response!
